src/components/ProfileForm.vue

```diff
@@ -0,0 +1,237 @@
+    <div class="form-control">
+        <label> First Name </label>
+        <input 
+        v-model="firstName"
+        type="text"
+        name="text"
+        />
+    </div>
+    
+    <div class="form-control">
+        <label> Last Name </label>
+        <input 
+        v-model="lastName"
+        type="text"
+        name="Last"
+        />
+    </div>
+    <div class="form-control">
+          <label>Age</label>
+          <input 
+        v-model="age"
+         type="text"
+         name="text"
+         />
+        </div>
+    <div class="form-control">
+        <label> Address </label>
+        <input 
+        v-model="address"
+        type="text"
+        name="text"
+        />
+    </div>
+    <div class="form-control">
+        <label> Gender </label>
+        <input 
+        v-model="gender"
+        type="text"
+        name="text"
+        />
+    </div>
+
+    <div class="save-form">
+          <button type="submit"> 
+            {{ sending ? 'saving...' : 'Save Form '}}
+        </button>
+    </div>
+  </form>
+  
+    </div>
+<!-- </div> -->
+</template>
+
+
+<script>
+
+
+export default {
+    name: "ProfileForm",
+
+    data() {
+        return {
+            firstName: "",
+            lastName: "",
+            age: "",
+            address: "",
+            gender: "",
+            sending: false,
+           
+            
+        };
+    },
+    methods : {
+        async submit(e) {
+            e.preventDefault();
+            
+            console.log('form submitted')
+            if (this.firstName === "" || this.lastName === "" || this.age === "" || this.address === "" || this.gender === "") {
+                this.$toast.warning("Please enter your details correctly!");
+            }else{
+                this.sending = true
+                const newProfile =  {
+                    firstname: this.firstName,
+                    lastname: this.lastName,
+                    age: this.age,
+                    address: this.address,
+                    gender: this.gender,
+                };
+                console.log (newProfile)
+                const response =  await fetch('https://ban-iot.herokuapp.com/api/profile', {
+                method: 'PUT',
+                headers: {
+                    'Authorization': `Bearer ${localStorage.getItem("token")}`,
+                    'Content-type': 'application/json',
+                },
+                body: JSON.stringify(newProfile)
+                })
+                const data = await response.json();
+                if (response.status === 200) {
+                console.log({ data });
+                this.$toast.success(data.message);
+                this.firstName = "";
+                this.lastName = "";
+                this.age = "";
+                this.address = "";
+                this.gender = "";
+                } else {
+                this.$toast.error("Error registering user");
```

Comment:
There is a message coming from response if not successful, use that instead of hardcoding it